src/services/rari.js

```diff
@@ -0,0 +1,217 @@
+import RARI_FUND_PROXY_CONTRACT_ABI from 'abi/rariFundProxy.json';
+import ERC20_CONTRACT_ABI from 'abi/erc20.json';
+
+
+export const getRariFundBalanceInUSD = async () => {
+  const rariContract = getContract(
+    getEnv().RARI_FUND_MANAGER_CONTRACT_ADDRESS,
+    RARI_FUND_MANAGER_CONTRACT_ABI,
+  );
+  if (!rariContract) return 0;
+  const balanceBN = await rariContract.callStatic.getFundBalance()
+    .catch((error) => {
+      reportErrorLog("Rari service failed: Can't get Rari fund balance", { error });
+      return 0;
+    });
+  return parseFloat(utils.formatUnits(balanceBN, 18));
+};
+
+export const getAccountDepositInUSD = async (accountAddress: string) => {
+  const rariContract = getContract(
+    getEnv().RARI_FUND_MANAGER_CONTRACT_ADDRESS,
+    RARI_FUND_MANAGER_CONTRACT_ABI,
+  );
+  if (!rariContract) return 0;
+  const balanceBN = await rariContract.callStatic.balanceOf(accountAddress)
+    .catch((error) => {
+      reportErrorLog("Rari service failed: Can't get user account deposit in USD", { error });
+      return 0;
+    });
+  return parseFloat(utils.formatUnits(balanceBN, 18));
+};
+
+export const getAccountDepositInRSPT = async (accountAddress: string) => {
+  const rariContract = getContract(
+    getEnv().RSPT_TOKEN_ADDRESS,
+    ERC20_CONTRACT_ABI,
+  );
+  if (!rariContract) return 0;
+  const balanceBN = await rariContract.balanceOf(accountAddress)
+    .catch((error) => {
+      reportErrorLog("Rari service failed: Can't get user account deposit in RSPT", { error });
+      return 0;
+    });
+  return parseFloat(utils.formatUnits(balanceBN, 18));
+};
+
+export const getUserInterests = async (accountAddress: string) => {
+  const userBalanceUSD = await getAccountDepositInUSD(accountAddress);
+  const userBalanceRSPT = await getAccountDepositInRSPT(accountAddress);
+  if (!userBalanceRSPT || !userBalanceUSD) return null;
+
+  /* eslint-disable i18next/no-literal-string */
+  const query = `{
+    transfersOut: transfers(where: {from: "${accountAddress}"}) {
+      amount
+      amountInUSD
+      timestamp
+    }
+    transfersIn: transfers(where: {to: "${accountAddress}"}) {
+      amount
+      amountInUSD
+      timestamp
+    }
+  }
+  `;
+  /* eslint-enable i18next/no-literal-string */
+  const transactions = await callSubgraph(getEnv().RARI_SUBGRAPH_NAME, query);
+  if (!transactions) return null;
+
+  // From Rari docs:
+  // "Get my interest accrued: Subtract total deposits and transfers in (in USD) and add total withdrawals
+  // and transfers out (in USD) from uint256 RariFundManager.balanceOf(address account)."
+  //
+  // But in order to calculate the interest percentage we "reset" the interests gained on the last transfer
+  // Aave calculates interest in the same way
+
+  const allTransactions = [...transactions.transfersIn, ...transactions.transfersOut].sort((a, b) => {
+    if (a === b) return 0;
+    if (a < b) return 1;
+    return -1;
+  });
```

Comment:
Apologies for not catching it earlier: since only the most recent transaction is used, there's no need for sorting, `maxBy` (already installed) should be enough.